README.md

```diff
@@ -59,32 +63,139 @@ let deleteFile containerName blob  =
                     |> Blob.execAsync<bool>
     result
+	
+let deleteTestJsonSnapshots =
+	async {
+        let! result = deleteSnapshots "container" "test.json"
+        match result.Value with
+        | true ->  Console.WriteLine "Success"
+        | false ->  failwith "Snapshots should be deleted"
+    } |> Async.RunSynchronously	
 ```
 
 ### Check if blob exists
 
 ```fsharp
-let existsFile containerName blob  =
+let existsFile containerName blob =
     let connString = "UseDevelopmentStorage=true"
     let result = connString
                     |> Blob.fromConnectionString
                     |> Blob.container containerName
                     |> Blob.exists blob
                     |> Blob.execAsync<bool>
     result
+	
+let checkTestJsonExists =
+	async {
+        let! result = existsFile "container" "test.json"
+        match result.Value with
+        | true ->  Console.WriteLine "Blob exists"
+        | false ->  failwith "Blob not found"
+    } |> Async.RunSynchronously  
+```
+
+### Retreive blob properties / metadata
+
+To learn more about properties and metadata please consult the documentation: 
+[Manage blob properties and metadata with .NET](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-properties-metadata?tabs=dotnet)
+
+```fsharp
+let getProperties containerName blob =
+    let connString = "UseDevelopmentStorage=true"
+    let result = connString
+                    |> Blob.fromConnectionString
+                    |> Blob.container containerName
+                    |> Blob.getProperties blob
+                    |> Blob.execAsync<BlobProperties>
+    result
+	
+let getTestJsonPropertiesAndMetadata =
+	async {
+        let! blobProperites = getProperties "container" "test.json"  
+        let properties = blobProperites.Value
+        Console.WriteLine "Properties:"
+        Console.WriteLine ("  Content Type: " + properties.ContentType)
+        Console.WriteLine ("  Content Length: " + properties.ContentLength.ToString())
+        let metadata = blobProperites.Value.Metadata
+        Console.WriteLine "Metadata:"
+        metadata.Keys |> Seq.iter (fun k -> printfn "  \"%s\" : \"%s\"" k (metadata.Item k))            
+    } |> Async.RunSynchronously
+```
+
+### Set / Update Properties
+
+Note that when you wish to change a property you need to set all other values in
+the property object by first retrieving them by calling get properties.
+
+```fsharp
+let setProperties containerName blob properties =
+    let connString = "UseDevelopmentStorage=true"
+    let result = connString
+                    |> Blob.fromConnectionString
+                    |> Blob.container containerName
+                    |> Blob.setProperties blob properties
+                    |> Blob.execAsync<BlobInfo>
+    result
+	
+let setTestJsonProperties =
+    async {
+        let! blobProperites = getProperties "container" "test.json"  
+        let properties = new Azure.Storage.Blobs.Models.BlobHttpHeaders()
```

Comment:
Please create a custom record type for blob properties and translate to Azure API in the operation